### PR TITLE
default shipping methods order by position

### PR DESCRIPTION
**Pull request by ehibes**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12 or 1.13                   |
| Bug fix?        | yes                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets |                       |
| License         | MIT                                                          |

I propose this modification, in order that the default shipping method is selected according to the position of the methods in the admin, even when the customer does not have an address yet.

### CODE REVIEW

**Comment by Rafikooo** [comment]
Hello @ehibes ,

Thank you for your contribution. First, please rebase the PR. Second, it would be great if you would add a behat test that covers such a feature 💃 
Another question - https://github.com/Sylius/Sylius/pull/14940/files#diff-40553a26e7e394962e1941848b0b41fd720f9edcca361f48c1062632f5355565R42-R43
What's the purpose of this join? 🤔

**Comment by ehibes** [comment]
Hi @Rafikooo, I added bug related behat scenario for @ui and @api. 
I removed the join, it was a bad copy/paste.

File: features/cart/shopping_cart/resolving_default_shipping_method_in_cart_summary_based_on_method_position.feature [hunk 1]
```diff
@@ -0,0 +1,17 @@
+@shopping_cart
+Feature: Viewing a cart summary with correct default shipping method based on variant applied
+    In order to see details about my order
+    As a Visitor
+    I want to be able to see my cart summary with the correct shipping method based on variants
```
Comment (NoResponseMate) [inline features/cart/shopping_cart/resolving_default_shipping_method_in_cart_summary_based_on_method_position.feature]:
```suggestion
Feature: Viewing a cart summary with the correct default shipping method
    In order to see details about my order
    As a Visitor
    I want see my cart summary with the correct shipping method based on its position
```

**Review by NoResponseMate — DISMISSED** [review]

**Review by NoResponseMate — APPROVED** [review]

**Review by jakubtobiasz — APPROVED** [review]

**Review by Rafikooo — APPROVED** [review]

File: features/cart/shopping_cart/resolving_default_shipping_method_in_cart_summary_based_on_method_position.feature [hunk 2]
```diff
@@ -0,0 +1,17 @@
+@shopping_cart
+Feature: Viewing a cart summary with the correct default shipping method
+    In order to see details about my order
+    As a Visitor
+    I want see my cart summary with the correct shipping method based on its position
```
Comment (coldic3) [inline features/cart/shopping_cart/resolving_default_shipping_method_in_cart_summary_based_on_method_position.feature:5]:
```suggestion
    I want to see my cart summary with the correct shipping method based on its position
```

File: src/Sylius/Behat/Context/Setup/ShippingContext.php [hunk 3]
```diff
@@ -143,6 +143,30 @@ public function theStoreAllowsShippingMethodWithNameAndPosition($name, $position
         $this->saveShippingMethod($shippingMethod);
     }
 
+    /**
+     * @Given /^the store(?:| also) allows shipping with "([^"]+)" at position (\d+) with ("[^"]+") fee$/
+     */
+    public function theStoreAllowsShippingMethodWithNameAndPositionAndFee($name, $position, $fee): void
```
Comment (coldic3) [inline src/Sylius/Behat/Context/Setup/ShippingContext.php:149]:
```suggestion
    public function theStoreAllowsShippingMethodWithNameAndPositionAndFee(string $name, int $position, $fee): void
```

**Review by coldic3 — APPROVED** [review]

**Comment by jakubtobiasz** [comment]
Thank you, @ehibes!

**Pull request merged**